### PR TITLE
Fix in BackoffSupervisor.calculateDelay - it no longer throws excepti…

### LINKED ISSUE #24419: Bug: BackoffSupervisor throws exception when it calculates delay
```scala
BackoffSupervisor.calculateDelay(
  restartCount = 29,
  minBackoff = 5.minutes,
  maxBackoff = 5.minutes,
  randomFactor = 0d)
``` 
throws `IllegalArgument exception`, as it tries to construct duration with nanos > Long.MaxValue.

I encountered this issue when using `RestartSource`
```scala
RestartSource.withBackoff(
  minBackoff = 5.minutes,
  maxBackoff = 5.minutes,
  randomFactor = 0d)(...))
```
After few restarts `RestartSource` terminates (which it shouldn't) with a failure:
```
java.lang.IllegalArgumentException: trying to construct too large duration with 1.207959552E19ns
	at scala.concurrent.duration.Duration$.fromNanos(Duration.scala:126)
	at scala.concurrent.duration.FiniteDuration.$times(Duration.scala:629)
	at akka.pattern.BackoffSupervisor$.calculateDelay(BackoffSupervisor.scala:158)
	at akka.stream.scaladsl.RestartWithBackoffLogic.scheduleRestartTimer(Restart.scala:359)
	at akka.stream.scaladsl.RestartWithBackoffLogic$$anon$8.onUpstreamFinish(Restart.scala:289)
	at akka.stream.stage.GraphStageLogic$SubSinkInlet.$anonfun$_sink$1(GraphStage.scala:1263)
	at akka.stream.stage.GraphStageLogic$SubSinkInlet.$anonfun$_sink$1$adapted(GraphStage.scala:1255)
	at akka.stream.impl.fusing.GraphInterpreter.runAsyncInput(GraphInterpreter.scala:447)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:465)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:560)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:742)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$shortCircuitBatch(ActorGraphInterpreter.scala:732)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:758)
	at akka.actor.Actor.aroundReceive(Actor.scala:517)
	at akka.actor.Actor.aroundReceive$(Actor.scala:515)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:667)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:527)
	at akka.actor.ActorCell.invoke(ActorCell.scala:496)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
	at akka.dispatch.Mailbox.run(Mailbox.scala:224)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

I'll submit relevant tests with my fix proposal shortly.

### PR DESCRIPTION
**Pull request by pwliwanow**
Fixes #24419

### CODE REVIEW

**Comment by raboof** [comment]
OK TO TEST

File: akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala [hunk 1]
```diff
@@ -152,13 +152,14 @@ object BackoffSupervisor {
...
     randomFactor: Double): FiniteDuration = {
     val rnd = 1.0 + ThreadLocalRandom.current().nextDouble() * randomFactor
-    if (restartCount >= 30) // Duration overflow protection (> 100 years)
-      maxBackoff
-    else
-      maxBackoff.min(minBackoff * math.pow(2, restartCount)) * rnd match {
-        case f: FiniteDuration ⇒ f
-        case _                 ⇒ maxBackoff
-      }
+    val calculatedDuration = try maxBackoff.min(minBackoff * math.pow(2, restartCount)) * rnd catch {
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala]: Maybe `Try(maxBackoff.min(minBackoff * math.pow(2, restartCount)) * rnd).getOrElse(maxBackoff)`?

**Review by ktoso — APPROVED** [review]
Looks good to me, thanks!

File: akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala (hunk 1 above)
Comment (pwliwanow) [reply akka-actor/src/main/scala/akka/pattern/BackoffSupervisor.scala]: Yes, it's nicer, I'll change it

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**